### PR TITLE
All claims 23 mg

### PR DESCRIPTION
**Pull request by markgreenburg**
## Description
Adds more options to the follow-up questions for new disabilities.

_Note:_ This PR doesn't include the disability name in the title of the follow-up questions, because this isn't supported by the forms library for string fields. Opened an [issue](https://github.com/usds/us-forms-system/issues/299) for this in the forms repo.

## Testing done
- [x] Tested locally
- [x] Unit tests

## Screenshots
First option / new primary disability:
screen shot 2018-10-09 at 12 05 27 pm

-----

Second option / new secondary disability:
screen shot 2018-10-09 at 12 05 53 pm

-----

Third option / new worsened disability:
screen shot 2018-10-09 at 12 06 18 pm

-----
Fourth option / new va mistreatment disability:
screen shot 2018-10-09 at 12 06 52 pm

-----

PTSD:
screen shot 2018-10-09 at 12 05 19 pm

## Acceptance criteria
- [x] Sorts PTSD-related disabilities to the end of the flow
- [x] Updates follow-up options per content in attached ticket
- [x] Updates follow-up questions for each follow-up option in the ticket
- [x] All follow-up questions conditionally required based on option selected
- [x] PTSD gets its own info page, PTSD form to be tackled in another PR

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by peggygannon — COMMENTED** [review]
@markgreenburg Hi! Screenshots look good..(I have no edits!). However, i didn't see a screenshot for the 
- Service-connected disability pop up box
- Summary of disabilities screen 
PTSD claims
{ INSERT BAH 0781 FLOW HERE }
X of Y Disabilities
Summary of disabilities
Below is the list of disabilities you're claiming in this application. If a disability is missing from the list, please go back one screen and add it.
{list of disabilities}

File: src/applications/disability-benefits/all-claims/config/form.js [hunk 1]
```diff
@@ -182,11 +183,40 @@ const formConfig = {
...
           arrayPath: 'newDisabilities',
           uiSchema: newDisabilityFollowUp.uiSchema,
           schema: newDisabilityFollowUp.schema,
         },
+        // Consecutive `showPagePerItem` pages that have the same arrayPath
+        // will force each item in the array to be evaluated by both pages
+        // before the next item is evaluated (e.g., if PTSD was entered first,
+        // it would still show first even though the first page was skipped).
+        // This break between the two `showPagePerItem`s ensures PTSD is sorted
+        // behind non-PTSD conditions in the form flow.
```
Comment (cvalarida) [inline src/applications/disability-benefits/all-claims/config/form.js]: Great comment!

File: src/applications/disability-benefits/all-claims/config/form.js [hunk 2]
```diff
@@ -182,11 +183,40 @@ const formConfig = {
           title: formData => getDisabilityName(formData.condition),
           path: 'new-disabilities/follow-up/:index',
           showPagePerItem: true,
-          itemFilter: item => !ptsdDisabilityIds.has(item.diagnosticCode),
+          itemFilter: item =>
+            item.condition
+              ? !item.condition.toLowerCase().includes(PTSD)
```
Comment (cvalarida) [inline src/applications/disability-benefits/all-claims/config/form.js]: Alternatively: `itemFilter: item => item.condition && !item.condition.toLowerCase().includes(PTSD)`

File: src/applications/disability-benefits/all-claims/config/form.js [hunk 3]
```diff
@@ -182,11 +183,40 @@ const formConfig = {
...
+          uiSchema: {},
+          schema: { type: 'object', properties: {} },
+        },
+        newPTSDFollowUp: {
+          title: formData => getDisabilityName(formData.condition),
+          path: 'new-disabilities/follow-up/ptsd/:index',
+          showPagePerItem: true,
+          itemFilter: item =>
+            item.condition
+              ? item.condition.toLowerCase().includes(PTSD)
```
Comment (cvalarida) [inline src/applications/disability-benefits/all-claims/config/form.js]: Alternatively: `itemFilter: item => item.condition && item.condition.toLowerCase().includes(PTSD)`

**Review by cvalarida — APPROVED** [review]

**Comment by markgreenburg** [comment]
Thanks for catching that @peggygannon - below are screenshots of the modal and the summary page.
The PTSD follow-up page with the 781 questions needs to be added during the integration step, so I've left it off this PR. The PTSD info page is still in here, though, with screenshots in the original PR comment.

Disabilities Summary:
screen shot 2018-10-09 at 4 43 09 pm

-----

Service-connected modal:
screen shot 2018-10-09 at 3 38 13 pm

**Comment by peggygannon** [comment]
@markgreenburg Thanks for posting the other screenshots. LGTM

**Pull request merged**